src/iconv.jl

```diff
@@ -57,6 +57,12 @@ type StringDecoder{S<:IO} <: IO
     skip::Int
 end
 
+# This is called during GC, just make sure C memory is returned, don't throw errors
```

Comment:
Good idea, I planned to do something about these annoying warnings.

But please use an `if` instead of `&&` and `;`., and `nothing` instead of `Void()`.